advice_for_future_giraf/process_advice.md

```diff
@@ -0,0 +1,38 @@
+# Advice for the Process of Future Giraf Projects
+This document provides some advice for the future process groups of the Giraf project. This advice is compiled by the process group of the 2019 Giraf project.
+
+## Do Use Full-Stack Teams
+During the Giraf 2019 project we chose to use a full-stack approach instead of the commonly used specialist approach. With the specialist approach each group is assigned a specific part of the system that they work on as a sole focus. E.g. two groups could be server groups, one could be back end, etc. With this approach each feature (user story) is potentially developed by many groups as they each work on different parts of the system to solve the issue. <br>
+With the full-stack approach each group develops full features (user stories) from top to bottom. This approach has many benefits such as:
+
+* Every group gets to see more of the code because they implement full features.
+* Less pipelining is needed as features are not made in parts by different groups.
+* People are motivated to finish their tasks because they can see their implementations immediately in the application.
+
+having used the full stack approach for this year's Giraf project, we believe that it is a good fit for this kind of semester project. We would therefore urge future groups to keep using this approach and not go back to the specialist approach.<p>
+This approach does have downsides that need to be managed still. One downside is that people collaborate less across groups, because every group works on separate user stories. This should be taken into account when using the full-stack approach. However, most of the downsides are not intrinsic to the approach itself and would also be a factor with the specialist approach.<br>
+Be careful to consider the following points when you think out the process for your own project:
+
+* How do you find dependencies between user stories? 
+    + How do you handle dependencies once discovered?
+* How do you make sure that one group's work is not made obsolete by another's? (Especially when the two groups are working in the same files.)
+
+We found that handling dependencies needs constant work with our process. It is quickly forgotten by individual groups that their work might affect others or that others might depend on one user story to implement their own if people are not made aware of this constantly.
+
+## Do Mention Everything Explicitly
+Throughout the 2019 Giraf project we found that the only way to make sure that every group followed the process correctly was to tell them every part of it (sometimes repeatedly). While this might sound trivial, there are a lot of small things that can be seen as implicit to some people and if the process group finds something to be self-evident they might not mention it. However small any part of the process might be, it should always be mentioned explicitly to everyone in the project.<br>
```

Comment:
```suggestion
Throughout the 2019 Giraf project we found that the only way to make sure that every group followed the process correctly was to tell them every part of it (sometimes repeatedly). While this might sound trivial, there are a lot of small things that can be seen as implicit to some people and if the process group finds something to be self-evident, they might not mention it. However small any part of the process might be, it should always be mentioned explicitly to everyone in the project.<br>
```